File: el_aventurero_front/src/app/core/models/arqueo-caja/create-arqueo.dto.ts
import { DetalleConteoModel } from './arqueo-caja.model';

export interface CreateArqueoDto {
  fecha: string;
  saldo_inicial: number;
  efectivo_real: number;
  observaciones?: string;

  // Detalle del conteo (aplanado para el backend)
  billetes_100000: number;
  billetes_50000: number;
  billetes_20000: number;
  billetes_10000: number;
  billetes_5000: number;
  billetes_2000: number;
  billetes_1000: number;
  monedas_1000: number;
  monedas_500: number;
  monedas_200: number;
  monedas_100: number;
  monedas_50: number;
}

File: el_aventurero/src/main/java/com/cloud_technological/el_aventurero/dto/arqueo_caja/CreateArqueoDto.java
package com.cloud_technological.el_aventurero.dto.arqueo_caja;

import java.math.BigDecimal;
import java.time.LocalDate;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CreateArqueoDto {
    private LocalDate fecha;
    private BigDecimal saldo_inicial;
    private BigDecimal efectivo_real;
    private String observaciones;
    
    // Detalle del conteo
    private Integer billetes_100000;
    private Integer billetes_50000;
    private Integer billetes_20000;
    private Integer billetes_10000;
    private Integer billetes_5000;
    private Integer billetes_2000;
    private Integer billetes_1000;
    private Integer monedas_1000;
    private Integer monedas_500;
    private Integer monedas_200;
    private Integer monedas_100;
    private Integer monedas_50;
}